client/lib/url/omit-url-params.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { URL as URLString } from 'types';
+import { Falsy } from 'utility-types';
+
+/**
+ * Internal dependencies
+ */
+import { determineUrlType, URL_TYPE } from './url-type';
+import format from './format';
+
+const BASE_URL = 'http://__domain__.invalid';
+
+/**
+ * Removes given params from a url.
+ *
+ * @param  url URL to be cleaned
+ * @param  paramsToOmit The collection of params or single param to reject
+ * @return Url less the omitted params.
+ */
+export default function omitUrlParams( url: Falsy, paramsToOmit: string | string[] ): null;
```

Comment:
I'll mention that I was considering the following change, opening up the second type parameter because we really don't care what it is in the falsy url case:

```ts
export default function omitUrlParams( url: Falsy, paramsToOmit: any ): null;
```

While that nicely expresses "falsy urls return null", it seems to complicate things in other ways 🙂 

For signatures like these, as much consistency as possible in the signatures (like you now have) seems best 👍

Reply:
👍